### PR TITLE
Mg dumper

### PR DESCRIPTION
**Pull request by mgratti**
This PR implements the additions that were done for the initial studies on the Graph Networks:

`Recosimdumper.cc`:
- added possibility to save actual PFRechits. But in truth, the option saveEBPFRechits = False in the cfg file. So there is no change

`prodHelper.py`:
- added another channel, named "overlap", to shoot photons close to each other 
- added option to set the number of particles manually, but the default is maintained
- added option to submit when running the production helper

New Dumper, called `RecosimdumperPF.cc` and corresponding cfg files. Compared to `Recosimdumper.cc`:
- all information about deep superclusters and retuned superclusters removed
- all truth information about superclusters removed
- normal superclusters information saved , including the shower shapes
- simhit information organised in a better way

@amlyon : after this PR, I think we could keep this version of RecoSimStudies for our own developments, should there be any.

### CODE REVIEW

File: Dumpers/python/Cfg_RecoSimDumper_cfi.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
     saveSimhits                       = cms.bool(True),  #save simHits information
     saveRechits                       = cms.bool(True),  #save recHits information
     savePFRechits                     = cms.bool(True),  #save pfRecHits information
+    saveEBPFRechits                   = cms.bool(False),
```
Comment (amlyon) [inline Dumpers/python/Cfg_RecoSimDumper_cfi.py:41]: @mgratti 

Out of curiosity, can you confirm if my understanding is correct:

- savePFRechits: saves the pfRecHits information for both EE and EB
- saveEBPRRecHits: saves the pfRecHits information only for EB (which is only what you needed, as you run the GNN study on barrel only?)
- --> if both options are set to true, there would be duplicated information for the barrel (?), therefore it is recommended to set only one to true at a time ?

File: Dumpers/python/Cfg_RecoSimDumper_cfi.py [hunk 2]
```diff
@@ -38,6 +38,7 @@
     saveSimhits                       = cms.bool(True),  #save simHits information
     saveRechits                       = cms.bool(True),  #save recHits information
```
Comment (amlyon) [inline Dumpers/python/Cfg_RecoSimDumper_cfi.py:39]: Minor comment/question

Do you actually think that some options could be turned to False (such as for instance saveSimHits and saveRechits), for the sake of having slimmer dumped_files? 

Or do those flags need to be set to true in order to compute higher-order variables (PFClusters et co.) ?

File: Dumpers/test/ECALproductionHelper/prodHelper.py [hunk 3]
```diff
@@ -58,6 +58,7 @@ def getOptions():
   parser.add_argument('--dosavehome', dest='dosavehome', help='save in home, otherwise save to SE', action='store_true', default=False)
   parser.add_argument('--doskipdumper', dest='doskipdumper', help='do not run the dumper at the end', action='store_true', default=False)
   parser.add_argument('--dodumperonly', dest='dodumperonly', help='only run the dumper', action='store_true', default=False)
+  parser.add_argument('--dosubmit', dest='dosubmit', help='do submission', action='store_true', default=False)
```
Comment (amlyon) [inline Dumpers/test/ECALproductionHelper/prodHelper.py:61]: This is a good idea :)

**Comment by amlyon** [comment]
@mgratti 

Thanks a lot for the PR. Everything looks fine to me, so feel free to proceed to merging.

File: Dumpers/python/Cfg_RecoSimDumper_cfi.py (hunk 1 above)
Comment (mgratti) [reply Dumpers/python/Cfg_RecoSimDumper_cfi.py:41]: in reality the option savePFRechits saves unclustered PFRechit if I remember correctly, while saveEBPFRechits saves all EB PFrechits

File: Dumpers/python/Cfg_RecoSimDumper_cfi.py (hunk 2 above)
Comment (mgratti) [reply Dumpers/python/Cfg_RecoSimDumper_cfi.py:39]: mmm, my naive answer would be: if we kept them for all this time there was a reason. 
However, if I go back to the code, saveSimhits we can definitely set to false. 
saveRecHits needs to be true to calculate the shower shape variables. but the logic is flawed. I will push a commit to fix it

Comment (amlyon) [reply Dumpers/python/Cfg_RecoSimDumper_cfi.py:39]: Thanks a lot!

**Pull request merged**